Topics/Development_Process/Shell/intro_to_shell.md

```diff
@@ -0,0 +1,159 @@
+
+## Understanding File System Structure
+
+- **Root Directory:** Denoted by `/`, it's the top-level directory.
+- **Home Directory:** Denoted by `~`, it's the user's personal directory.
+- **Relative Path:** Specifies a location relative to the current directory.
+- **Absolute Path:** Specifies a location from the root directory.
+
+## Basic Navigation Commands
+
+- `pwd`: Print Working Directory. Shows your current directory.
+  - **Usage:** `pwd` - Displays the path of the current directory.
+
+- `ls`: Lists files and directories.
+  - **Usage:** `ls [options] [directory]`
+  - **Options:** `-l` for detailed view, `-a` to show hidden files.
+  - **Example:** `ls -la ~/Documents` - Lists all files in Documents with details.
+
+- `cd`: Change Directory. Moves to another directory.
+  - **Usage:** `cd [directory]`
+  - **Example:** `cd /var/www` - Moves to the `/var/www` directory.
+  - **Tip:** `cd ..` moves you up one directory.
+
+## Tips for Effective Navigation
+
+1. **Tab Completion:** Press the Tab key to auto-complete file and directory names.
+2. **History Navigation:** Use the Up and Down arrow keys to navigate through previously used commands.
+3. **File Paths:** Understand the difference between relative and absolute paths to navigate effectively.
+
+Mastering these commands is the first step towards proficiency in the Unix-like command-line environment.
+
+# File Manipulation
+
+Manipulating files is a frequent task in the command line. Here are some of the most commonly used commands for handling files:
+
+## Creating Files and Directories
+
+- `touch`: Creates a new empty file or updates the timestamp of an existing file.
+  - **Usage:** `touch [filename]`
+  - **Example:** `touch example.txt` - Creates a new file named `example.txt`.
+
+- `mkdir`: Creates a new directory.
+  - **Usage:** `mkdir [directory name]`
+  - **Example:** `mkdir new_folder` - Creates a new directory named `new_folder`.
+
+## Copying and Moving Files
+
+- `cp`: Copies files or directories.
+  - **Usage:** `cp [source] [destination]`
+  - **Options:** `-r` for recursive copying (directories), `-i` for interactive mode.
+  - **Example:** `cp file.txt ~/Documents/` - Copies `file.txt` to the Documents directory.
+
+- `mv`: Moves or renames files or directories.
+  - **Usage:** `mv [source] [destination]`
+  - **Example:** `mv oldname.txt newname.txt` - Renames `oldname.txt` to `newname.txt`.
+
+## Deleting Files and Directories
+
+- `rm`: Removes files or directories.
+  - **Usage:** `rm [file or directory]`
+  - **Options:** `-r` for recursive deletion (directories), `-f` for force deletion without prompt.
+  - **Example:** `rm -rf old_folder/` - Forcefully removes `old_folder` and its contents.
+
+## Tips for Safe File Manipulation
+
+1. **Double-check file names and paths before executing commands**, especially with `rm`.
+2. **Use wildcards (`*`) carefully**, as they can affect multiple files.
```

Comment:
Have you mentioned what wildcards do? You probably can give an example of wild cards in another sub-section. 
Or just remove this line altogether.